File: C++/programas/POO-Constructores y Destructores/POO-Constructores y Destructores/Triangulo.cpp
#include "Triangulo.h" //Cabecera de clase Triangulo
#include <iostream> //Cabecera para entrada y salida de datos.

Triangulo::Triangulo(float base, float altura) { //Inicialización de atributos protegidos por medio de constructor.
	Base = base;
	Altura = altura;
}

Triangulo::~Triangulo() { //Despliega un mensaje por eliminación de objeto.
	std::cout << "\nSe ha eliminado el objeto Triangulo.\n";
}

//Función que regresa el área de un triángulo.
float Triangulo::Area() {
	return a = (Base * Altura) / 2;
}

//Función que regresa el perímetro de un triángulo.
float Triangulo::Perimetro() {
	if (Base != Altura)
		return p = (Altura * 2) + Base;
	return p = Base * 3;
}

//Procedimiento que despliega información de un triángulo.
void Triangulo::Resultado() {
	std::cout << "\n--Datos del triángulo--\n\n";
	std::cout << "Base del triángulo: " << Base << "u.\n";
	std::cout << "Altura del triángulo: " << Altura << "u.\n";
	std::cout << "Área del triángulo: " << a << "u^2.\n";
	std::cout << "Perímetro del triángulo: " << p << "u.\n\n";
}

File: Java/programas/POO-Constructores y Destructores/src/Triangulo.java
public class Triangulo extends Figuras{
    //Constructor que inicializa valores a los atributos públicos de la clase padre por medio de "super.".
    public Triangulo(float base, float altura){
        super.Base = base;
        super.Altura = altura;
    }

    //Función que devuelve el perímetro del triángulo utilizando los atributos de la clase padre.
    public float Area(){
        return super.a = (super.Base * super.Altura) / 2;
    }

    //Función que devuelve el perímetro del triángulo utilizando los atributos de la clase padre.
    public float Perimetro(){
        if (super.Base != super.Altura){
            return super.p = (super.Altura * 2) + super.Base;
        }
        return super.p = super.Base * 3;
    }

    //Procedimiento que despliega la base, altura, área y perímetro de un triángulo.
    public void Resultado(){
        System.out.println("---Datos del triángulo---\n");
        System.out.println("Base del triángulo: " + super.Base);
        System.out.println("Altura del triángulo: " + super.Altura);
        System.out.println("Área del triángulo: " + super.a + "u^2");
        System.out.println("Perimetro del triángulo: " + super.p + "u");
    }

    //Destructor no necesario.
}